controllers/awsmachine_controller.go

```diff
@@ -564,6 +568,20 @@ func (r *AWSMachineReconciler) reconcileLBAttachment(machineScope *scope.Machine
 	}
 
 	elbsvc := elb.NewService(clusterScope)
+
+	// In order o prevent sending request to a "not-ready" control plane machines, it is required to remove the machine
+	// from the ELB as soon as the machine gets delete or when the machine is detected or not running
+	// Nb. this is a reinforcement to the ELB health check, which blocks traffic only after a timeout is reached.
```

Comment:
I'd remove this last bit, it's more of a separate use case. I see health check to be useful when a machine goes unhealthy for reasons unrelated to intended operations on our objects. The points above explain the reasoning very well. 

Basically, I don't want to suggest there is something wrong with the current health checks, because those are meant to cover a different set of use cases